### PR TITLE
Update docstrings, tests, comments

### PR DESCRIPTION
**Pull request by jcampbell**
- Default value for run_id when invoking a validation_operator
- Small test fixes and improvements
- Small doc fixes and improvements

### CODE REVIEW

**Comment by jcampbell** [comment]
@cla-bot check

**Comment by jcampbell** [comment]
The cla-bot has been summoned, and re-checked this pull request!

**Comment by jcampbell** [comment]
@cla-bot check

File: great_expectations/data_context/data_context.py [hunk 1]
```diff
@@ -460,17 +457,27 @@ def get_batch(self, batch_kwargs, expectation_suite_name, data_asset_type=None,
...
+        if isinstance(batch_kwargs, dict):
+            batch_kwargs = BatchKwargs(batch_kwargs)
+
+        if not isinstance(batch_kwargs, BatchKwargs):
+            raise ge_exceptions.BatchKwargsError("BatchKwargs must be a BatchKwargs object or dictionary.")
+
+        if not isinstance(expectation_suite_name, (ExpectationSuiteIdentifier, string_types)):
+            raise ge_exceptions.DataContextError("expectation_suite_name must be an ExpectationSuiteIdentifier or "
+                                                 "string.")
+
```
Comment (Aylr) [inline great_expectations/data_context/data_context.py:488]: These make good sense. Are they covered in tests?

File: great_expectations/types/configurations.py [hunk 2]
```diff
@@ -1,14 +1,3 @@
...
-# from ruamel.yaml import YAML, yaml_object
-#
-#
-# # from great_expectations.types import AllowedKeysDotDict
-#
-# yaml = YAML()
-
-#
-# class Config(AllowedKeysDotDict):
-#     pass
```
Comment (Aylr) [inline great_expectations/types/configurations.py:11]: ❤️

File: great_expectations/types/configurations.py [hunk 3]
```diff
@@ -30,17 +19,3 @@ def module_name(self):
...
-#     #     "module_name",
-#     #     "class_name"
-#     # }
-#     # _required_keys = {
-#     #     "class_name"
-#     # }
-#     # _key_types = {
-#     #     "module_name": string_types,
-#     #     "class_name": string_types
-#     # }
```
Comment (Aylr) [inline great_expectations/types/configurations.py:46]: It's a red party!
red_party

**Review by Aylr — COMMENTED** [review]
lgtm w/ a question about testing of `BatchKwargError` states.

**Review by Aylr — APPROVED** [review]
sorry wrong radio button,. This was supposed to be approved

**Pull request merged**